datastore/src/main/java/com/openlattice/datastore/data/controllers/DataController.java

```diff
@@ -981,6 +987,33 @@ public long getEntitySetSize( @PathVariable( ENTITY_SET_ID ) UUID entitySetId )
         return dgm.getLinkedEntitySetBreakDown( entityKeyIdsOfEntitySets, authorizedPropertyTypesOfEntitySets );
     }
 
+    @Timed
+    @Override
+    @PostMapping(
+            path = SET_ID_PATH + "/" + ENTITY_KEY_ID_PATH + "/" + PROPERTY_TYPE_ID_PATH + "/" + DIGEST_PATH,
+            consumes = MediaType.APPLICATION_JSON_VALUE,
+            produces = MediaType.APPLICATION_JSON_VALUE
+    )
+    public URL downloadBinaryPropertyWithContentDisposition(
+            @PathVariable( ENTITY_SET_ID ) UUID entitySetId,
+            @PathVariable( ENTITY_KEY_ID ) UUID entityKeyId,
+            @PathVariable( PROPERTY_TYPE_ID ) UUID propertyTypeId,
+            @PathVariable( DIGEST ) String digest,
```

Comment:
Isn't `digest` the result of `PostgresDataHasher.hashObjectToHex`? As far as I know, we don't have access to the hash on the frontend; the neighbors request does not return it. 

How would the caller get the hash?

Reply:
it's part of the presigned URL that's returned. actually the whole segment of this path can be taken directly from there, since it uses the same fields in the same order.